source/builtin_macros/src/syntax.rs

```diff
@@ -5384,3 +5392,80 @@ fn check_verus_return_ident(
     }
     None
 }
+
+///
+fn check_async_fn_open_invariants(sig: &mut Signature) -> Option<Stmt> {
```

Comment:
This sort of check should be performed in Verus itself (e.g., `vir/src/well_formed.rs`)